web/src/components/deployments-detail-page/log-viewer/index.tsx

```diff
@@ -4,20 +4,39 @@ import {
   makeStyles,
   Toolbar,
   Typography,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+  Button,
 } from "@material-ui/core";
-import { Close } from "@material-ui/icons";
+import { Close, SkipNext } from "@material-ui/icons";
 import clsx from "clsx";
 import { FC, memo, useCallback, useState } from "react";
 import Draggable from "react-draggable";
 import { APP_HEADER_HEIGHT } from "~/components/header";
-import { useAppDispatch, useShallowEqualSelector } from "~/hooks/redux";
+import {
+  useAppDispatch,
+  useShallowEqualSelector,
+  useAppSelector,
+} from "~/hooks/redux";
 import { clearActiveStage } from "~/modules/active-stage";
-import { isStageRunning, selectById, Stage } from "~/modules/deployments";
+import {
+  isStageRunning,
+  selectById,
+  Stage,
+  StageStatus,
+  skipStage,
+  selectDeploymentStageIsSkipping,
+  updateSkippingState,
+} from "~/modules/deployments";
 import { selectStageLogById, StageLog } from "~/modules/stage-logs";
 import { Log } from "./log";
 
 const INITIAL_HEIGHT = 400;
 const TOOLBAR_HEIGHT = 48;
+const ANALYSIS_NAME = "ANALYSIS";
```

Comment:
```suggestion
const ANALYSIS_STAGE_NAME = "ANALYSIS";
```
nits, and caller place at L186